### DISCUSSION TITLE
How do I add compiler options?

### DISCUSSION THREAD

**Discussion by pepper-lees** [scalameta/nvim-metals#580] [Q&A]
Hi,

Metals Doctor is giving me this recommendation for some subprojects:
```
recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.
```
How do I add a compiler option for metals?

**Comment by ckipp01** [comment]
Hey @JL102, Metals should actually take care of this for you. Are you using sbt or Bloop as your build server? Could you include the full output of the doctor here? It might be as simple as reloading the build to make sure everything gets refreshed. Sometimes (especially with sbt as your build server) those options get added, but the build doesn't get reloaded correctly. This was fixed recently, but I don't think it's in the latest release of Metals yet.

**Reply by pepper-lees** [reply]
Sure, here's the full output.
The project is using sbt.
```
## Metals Info
  - Metals Server version: 0.11.12
  - Metals Java: 17.0.3-internal from N/A located at /home/bxeuser/chipyard/.conda-env/lib/jvm
  - Build server currently being used is sbt v1.8.2.

These are the installed build targets for this workspace. One build target corresponds to one classpath. For example, normally one sbt project maps to two build targets: main and test.

## Build Targets

### chipyardRoot
  - target type: Scala 2.13.8
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### chipyardRoot-test
  - target type: Scala 2.13.8
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### fpga_platforms
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### fpga_platforms-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### fpga_shells
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### fpga_shells-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### boom
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### boom-test
  - target type: Scala 2.13.10
  - compilation: ❌  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### chipyard
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### chipyard-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### constellation
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### constellation-test
  - target type: Scala 2.13.10
  - compilation: ❌  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### cva6
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### cva6-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### fft_generator
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### fft_generator-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### firechip
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### firechip-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### gemmini
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### gemmini-test
  - target type: Scala 2.13.10
  - compilation: ❌  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### hwacha
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### hwacha-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### ibex
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### ibex-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### icenet
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### icenet-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### async_mem
  - target type: Scala 2.13.10
  - compilation: ❌  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### async_mem-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### mempress
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### mempress-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### nvdla
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### nvdla-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sodor
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sodor-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### hardfloat
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### hardfloat-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### rocketMacros
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### rocketMacros-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### rocketchip
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### rocketchip-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sha3
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sha3-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sifive_blocks
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sifive_blocks-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sifive_cache
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### sifive_cache-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### testchipip
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### testchipip-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### tracegen
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### tracegen-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### chipyardRoot-build
  - target type: sbt 1.8.2
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ❌  - java: ⚠ 
  - recommendation: Diagnostics and debugging for sbt are not supported currently.

### firesim
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### firesim-scalaunidoc
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### firesim-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### firesim-testscalaunidoc
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### firesimLib
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### firesimLib-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### midas
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### midas-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### targetutils
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### targetutils-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 
  - recommendation: Add the compiler option "-P:semanticdb:sourceroot:/home/bxeuser/chipyard" to ensure code navigation works.

### firesim-build
  - target type: sbt 1.8.2
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ❌  - debugging: ❌  - java: ⚠ 
  - recommendation: Run 'Import build' to enable code navigation.

### tapeout
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### tapeout-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### iocell
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### iocell-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### cde
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### cde-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### dsptools
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### dsptools-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### rocket-dsp-utils
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

### rocket-dsp-utils-test
  - target type: Scala 2.13.10
  - compilation: ✅ 
  - diagnostics: ✅ 
  - interactive: ✅ 
  - semanticdb: ✅ 
  - debugging: ✅ 
  - java: ⚠ 

## Explanations

### Compilation status:
✅  - code is compiling
❌ - code isn't compiling, open problems tab to see detailed compilation errors

### Diagnostics:
✅  - diagnostics correctly being reported by the build server

### Interactive features (completions, hover):
✅  - supported Scala version

### Semanticdb features (references, renames, go to implementation):
✅  - build tool automatically creating needed semanticdb files
❌ - semanticdb not being produced

### Debugging (run/test, breakpoints, evaluation):
✅  - users can run or test their code with debugging capabilities
❌ - the tool does not support debugging in this target

### Java Support:
✅  - working non-interactive features (references, rename etc.)
❌ - missing semanticdb plugin, might not be added automatically by the build server (which is only done when using Bloop)
ℹ  - build target doesn't support Java files
```

**Reply by ckipp01** [reply]
Huh, it's interesting that this is only happening in certain modules. I tried to clone and setup Chipyard but had some issues as I couldn't get the submodules all set up. Is there any instructions on getting everything set up that I could look at to try and reproduce this?

**Reply by pepper-lees** [reply]
Yeah, Chipyard is a huge repo. There's instructions here: https://chipyard.readthedocs.io/en/stable/Chipyard-Basics/Initial-Repo-Setup.html

TL;DR / notes:
1. Need Conda-forge or Miniforge version 4.12.0 or higher
2. Install `conda-lock==1.4.0` then `conda activate base`
3. In the git repo, do `git checkout 1.9.1`
4. There's a bug causing the Firesim and Firemarshal submodules to not build properly, so to skip it in the setup process, do `./build-setup.sh riscv-tools -s 6 -s 7 -s 8 -s 9` instead of `./build-setup.sh riscv-tools`
5. Lastly, source `env.sh` before opening nvim - I'm not sure why I need to do that in order to get Metals to run properly, but if I don't, then it exits with code 127

**Reply by ckipp01** [reply]
So trying to follow this I get
```
~/Documents/scala-workspace/chipyard tags/1.9.1
base ❯ ./build-setup.sh riscv-tools -s 6 -s 7 -s 8 -s 9                                                                                                                      (base)
/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/pydantic/_internal/_config.py:257: UserWarning: Valid config keys have changed in V2:
* 'json_encoders' has been removed
  warnings.warn(message, UserWarning)
/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/pydantic/_internal/_config.py:257: UserWarning: Valid config keys have changed in V2:
* 'allow_mutation' has been removed
  warnings.warn(message, UserWarning)
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/bin/conda-lock", line 10, in <module>
    sys.exit(main())
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/conda_lock/conda_lock.py", line 1422, in install
    with _render_lockfile_for_install(
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/conda_lock/conda_lock.py", line 934, in _render_lockfile_for_install
    lock_content = parse_conda_lock_file(pathlib.Path(filename))
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/conda_lock/lockfile/__init__.py", line 137, in parse_conda_lock_file
    return lockfile_v1_to_v2(LockfileV1.parse_obj(content))
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/typing_extensions.py", line 2562, in wrapper
    return __arg(*args, **kwargs)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/pydantic/main.py", line 935, in parse_obj
    return cls.model_validate(obj)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/pydantic/main.py", line 480, in model_validate
    return cls.__pydantic_validator__.validate_python(
pydantic_core._pydantic_core.ValidationError: 1 validation error for Lockfile
package.390.optional
  Field required [type=missing, input_value={'dependencies': {}, 'has....whl', 'version': '6.0'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.0.1/v/missing
```

I'm not super familiar with the python ecosystem, so I'm not really sure how realistic this will be for me to reproduce.

**Reply by pepper-lees** [reply]
Hmm, someone else reported the same issue a couple hours before you sent that reply: https://github.com/ucb-bar/chipyard/issues/1441#issuecomment-1607767964

Could I ping you when someone finds a solution? It's an interesting coincidence to see two people with the same issue at the same time. I wonder if it's related to a recent release of Mambaforge.

**Reply by ckipp01** [reply]
Hey, sure, just no promises to when I'll get around to trying it again.

**Reply by pepper-lees** [reply]
All good, I appreciate you taking the time nevertheless!

**Reply by pepper-lees** [reply]
FYI: After solving #584, and checking the doctor again, I no longer see that compiler option recommendation, and it does appear that code navigation works, as far as I can see!
